### PR TITLE
Update Scripts that Launch MySQL Command Line Client

### PR DESCRIPTION
**Pull request by sureshc**
Update the Ruby scripts that launch the MySQL command line client to use the new set of `CDO.` database configuration settings that are populated from stack-specific AWS Secrets, that were provisioned via CloudFormation, using the cluster's writer and reader endpoints and/or RDS Proxy writer, reader, reporting endpoints as appropriate. This will be a first adoption of RDS Proxy and an intermediate step towards having our web application use RDS Proxy instead of ProxySQL.

Rename the scripts to use a hierarchical naming convention to make it easier to understand what the scripts do and to use tab completion to find the correct script.

- mysql-client-admin <-- admin-sql
- mysql-client-dashboard-writer <-- dashboard-sql-writer
- mysql-client-dashboard-reader <-- dashboard-sql
- mysql-client-dashboard-reporting <-- dashboard-reporting-sql
- mysql-client-pegasus-writer <-- pegasus-sql-writer
- mysql-client-pegasus-reader <-- pegasus-sql
- mysql-client-pegasus-reporting <-- pegasus-reporting-sql

Internal documents that need to be updated to reference the new script names:

- "How to access production MySQL databases"
- "Drone Continuous Integration System"
- "Connect adhoc to cloned DB"
- "Tricks and Tips for Code.org Engineers" 

## Testing story

`bundle exec rake adhoc:start RAILS_ENV=adhoc DATABASE=1 STACK_NAME=adhoc-mysql-client-scripts`

## Follow-up work

1. Consolidate these scripts into a single `bin/mysql-client` that defaults to the reporting RDS Proxy in production, the reader in test, and admin in development environments, which command line arguments to override those defaults, where needed.
2. Implement a CloudFormation custom Lambda Resource that to create/update/delete our applications' writer and reader SQL users and grant them privileges.
3.   Implement logic to configure Rails to use the new database client configuration settings and to conditionally connect to via RDS Proxy when running as a web application server.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/cloudformation/components/database.yml.erb [hunk 1]
```diff
@@ -65,6 +65,8 @@
     Properties:
       DBProxyName: !Sub "${AWS::StackName}-reporting"
       EngineFamily: MYSQL
+      # TODO: Remove this debug.
+      DebugLogging: true
```
Comment (cat5inthecradle) [inline aws/cloudformation/components/database.yml.erb]: delete or add deletion criteria?

**Review by cat5inthecradle — CHANGES_REQUESTED** [review]
Make sure engineers get a heads up on these renames. Consider requesting a review from Dev Tools Working group (has a github group)

This impacts lots of docs too, both in and out of the repo. For example, `SETUP.md` references `bin/dashboard-sql`.

**Comment by sureshc** [comment]
@code-dot-org/dev-tools-group Thoughts on the new names for these scripts that launch they MySQL client? My intention here is to:

-  give them names that explain what they actually do so we don't have to document that they exist to new engineers (we have a half dozen Google docs that tell engineers about these scripts)
- use a naming convention that ensures they are all displayed together when listed alphabetically
- use a hierarchical naming convention so that engineers can easily discover full range of options available to them

I'm considering combining this into a single script (`bin/mysql-client`) which would:

- Default to connecting as admin directly to the writer endpoint in development and adhoc environments
- Default to connecting to dashboard with the reader user to the reader proxy endpoint on test
- Default to connecting to dashboard with the reader user to the reporting proxy endpoint on production (reporting endpoint ensures long running queries don't impact production performance).
- Present interactive options to select a different option (writer user, writer endpoint, etc.) and/or accept command line arguments to override the above defaults.

**Comment by davidsbailey** [comment]
> * Present interactive options to select a different option (writer user, writer endpoint, etc.) and/or accept command line arguments to override the above defaults.

I like the proposal in general. If you go the single script option, I would personally advocate for command line options over interactive options, unless the interactive options can be avoided/overridden by specifying the corresponding command line options (don't get me started on how many adhocs I've started and then failed to press `y`).

**Review by Hamms — COMMENTED** [review]
Renames sound good to me, new secrets look right, and I like the plan for next steps!

Are we worried about any chicken-and-egg problems here? I know it was necessary for us to add the new options with placeholder values first, but It's not clear to me whether or not it's necessary for us to break up into multiple PRs the three steps of enabling the new options with `!StackSecret`, using them instead of the old options, and removing the old options.

**Comment by sureshc** [comment]
> Are we worried about any chicken-and-egg problems here? I know it was necessary for us to add the new options with placeholder values first, but It's not clear to me whether or not it's necessary for us to break up into multiple PRs the three steps of enabling the new options with `!StackSecret`, using them instead of the old options, and removing the old options.

Because the old `CDO.` configuration settings populated from `!Secrets` and the new `CDO.` settings populated from `!StackSecrets` are ONLY referenced by the MySQL client scripts, we don't have any circular-dependency issues with this change. But, yes, typically we would have to deploy the new config settings in one Pull Request, then a separate change to start using them, and then a third change to remove the old settings.

**Comment by sureshc** [comment]
@cat5inthecradle Are the changes you requested resolved?

**Review by cat5inthecradle — APPROVED** [review]

**Pull request merged**